code/game/machinery/washing_machine.dm

```diff
@@ -57,283 +118,307 @@
+	running = FALSE
+	if(!full)
+		bloody = FALSE
 	else
-		state = 4
+		playsound(src, 'sound/weapons/jug_filled_impact.ogg', 10)
 	update_icon()
+	sleep(5) // todo: replace this with a callback
+	if(stat & NOPOWER)
+		playsound(src, 'sound/machines/terminal_off.ogg', 45)
+	else
+		playsound(src, 'sound/machines/defib_success.ogg', 25)
 
-/obj/machinery/washing_machine/verb/climb_out()
-	set name = "Climb out"
-	set category = "Object"
-	set src in usr.loc
-
-	sleep(20)
-	if(state in list(1,3,6) )
-		usr.loc = src.loc
-
+/obj/machinery/washing_machine/proc/dyethis(var/obj/item/Item, wash_color, clothing = TRUE) // god this is nasty
+	// Things that are acceptable to dye
+	var/list/dyeablethings = list(/obj/item/clothing/under/plasmaman, /obj/item/clothing/head/helmet/space/plasmaman, /obj/item/clothing/under, /obj/item/clothing/gloves/color, /obj/item/clothing/shoes, /obj/item/clothing/mask/bandana, /obj/item/clothing/head/soft, /obj/item/clothing/head/beanie, /obj/item/clothing/accessory/scarf, /obj/item/bedsheet, /obj/item/storage/wallet/color, /obj/item/toy/carpplushie)
+	var/list/possible = list()
+	for(var/obj/item/Path in dyeablethings) // for every item in contents, check every it against every dyeablething, then check that for every subtype (god what the fuck)
+		if(istype(Item, Path))
+			for(var/T in typesof(Path))
+				var/obj/item/J = new T // for the love of god change the variable name
+				if(clothing)
+					var/obj/item/clothing/J2 = T // this one too
+					if(!initial(J2.dyeable))
+						continue
+				if((wash_color == J.item_color))
+					possible += J
+				qdel(J)
+			if(possible)
+				var/obj/item/Why = pick(possible)
+				to_chat(world, "[possible.Join(", ")]")
+				to_chat(world, "[Why]")
+				Item.item_state = Why.item_state
+				Item.icon_state = Why.icon_state
+				Item.item_color = Why.item_color
+				Item.name = Why.name
+				Item.desc = Why.desc
+				Item.desc += "\nThe colors seem slightly dodgy."
+				break
+			else
+				to_chat(world, "this is a bug")
 
 /obj/machinery/washing_machine/update_icon()
-	icon_state = "wm_[state][panel]"
+	if(bloody)
+		if(running)
+			state = 8
+		else
+			if(open)
+				state = 6
+			else
+				state = 7
+	else
+		if(running)
+			state = 5
+		else
+			if(full)
+				if(open)
+					state = 3
+				else
+					state = 4
+			else
+				if(open)
+					state = 1
+				else
+					state = 2
 
-/obj/machinery/washing_machine/attackby(obj/item/W as obj, mob/user as mob, params)
-	/*if(istype(W,/obj/item/screwdriver))
-		panel = !panel
-		to_chat(user, "<span class='notice'>you [panel ? </span>"open" : "close"] the [src]'s maintenance panel")*/
+	icon_state = "wm_[state]"
+	cut_overlays()
+	if(panel)
+		add_overlay("wires")
+	else
+		if(stat & NOPOWER)
+			if(running)
+				add_overlay("power0")
+		else
+			add_overlay("power1")
+
+/obj/machinery/washing_machine/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/reagent_containers/spray/cleaner) || istype(W, /obj/item/soap))
```

Comment:
what's the point of this check if people can alt-click the washer in an earlier proc to clean it without soap/cleaner?